apps/api/src/app/events/services/mail-service/handlers/msgraph.handler.ts

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { ChannelTypeEnum } from '@novu/shared';
+import { MSGraphAPIProvider } from '@novu/msgraph';
+import { BaseHandler } from './base.handler';
+import { ICredentials } from '@novu/dal';
+
+export class MSGraphHandler extends BaseHandler {
+  constructor() {
+    super('msgraph', ChannelTypeEnum.EMAIL);
+  }
+  buildProvider(credentials: ICredentials, from?: string) {
+    // null-assertion is disabled because we want to expose the error from the MS Graph API
+    this.provider = new MSGraphAPIProvider({
+      clientId: credentials.clientId!,
+      authority: credentials.applicationId!,
+      secret: credentials.apiKey!,
+      scopes: credentials.region!,
+      user: credentials.user!,
+      fromAddress: credentials.host!,
+      fromName: credentials.from!,
```

Comment:
```suggestion
      fromAddress: credentials.from!,
      fromName: credentials.senderName!,
```
Based in our credentials interface I consider these are the most appropriate for you to use.
https://github.com/novuhq/novu/blob/ce3af8529229eb5131c4b8d981addc439c50ad6d/apps/web/src/pages/integrations/IntegrationsStorePage.tsx#L142